bitcoin/src/pow.rs

```diff
@@ -234,6 +234,34 @@ impl Target {
     /// [`difficulty`]: Target::difficulty
     #[cfg_attr(all(test, mutate), mutate)]
     pub fn difficulty_float(&self) -> f64 { TARGET_MAX_F64 / self.0.to_f64() }
+
+    /// Computes the minimum valid [`Target`] threshold allowed for a block in which a difficulty
+    /// adjustment occurs. Note that the result is bounded by [`Target::MAX`].
+    ///
+    /// The difficulty can only decrease or increase by a factor of 4 max on each difficulty
+    /// adjustment period.
+    pub fn min_difficulty_transition_threshold(&self) -> Self {
+        let min = Self(self.0 >> 2);
+        if min > Self::MAX {
+            Self::MAX
+        } else {
+            min
+        }
```

Comment:
I'm not a fan of this, it reads funnily. If we are not upholding the invariant that `Self` is less than or equal to `Self::MAX` then we should fix that and not band aid over it here.

Reply:
I raised an issue: https://github.com/rust-bitcoin/rust-bitcoin/issues/1828

Reply:
Agree.

Reply:
o.O you can have difficulty < 1 for regtest and signet? What is the corresponding target value?

Reply:
Or do you mean that the conversion between difficulty and target is different on these networks....which would require we modify our conversion functions.

Reply:
Agree let's not hold this up.